CONTRIBUTING.md

```diff
@@ -0,0 +1,86 @@
+# Contributing Guidelines
+
+These guidelines are here to ensure that we maintain a clean, linear Git
+history. All contributions should be made in the form of pull requests (PRs).
+
+For quick reference about Git commands and concepts, see John's
+[notes on using Git][j-hui-yagg].
+
+[j-hui-yagg]: https://j-hui.com/pages/yagg
+
+## Rules
+
+- Collaborators all have write access, so be mindful of what others are working
+  on.
+
+  - Each PR should have at least one approving review before being merged in,
+    preferably from someone also responsible for the code that the PR affects
+    (though not someone who has also worked on the PR). This repo is configured
+    to require one approving review, though anyone may submit a review.
+
+  - The author of the PR should be the one to merge in the PR, unless there they
+    explicitly instruct someone else to do it.
+
+  - Do not push directly to `main`. This branch is already configured with push
+    protections to prevent this.
+
+- Write sensible, descriptive commit messages.
+
+  - Follow [this guide][git-style]: summarize in 50 characters or less (but
+    elaborate in the message body), in the imperative tense.
+
+  - Quick and dirty commit messages are fine in PRs, but make sure they are
+    squashed by the time they are merged into `main` or anyone else's branch.
+
+- Keep the `main` commit history clean.
+
+  - Always make sure to look at that your commits don't touch files
+
+  - If your PR contains a sequence of such commits, make sure to "Squash and
+    Merge", and provide a good commit message to summarize all the changes made.
+
+  - If you want to land multiple _clean_ commits onto `main`, make sure to clean
+    them up locally, then use "Rebase and Merge".
+
+  - Each commit should point to working tree that is in a sensible state (i.e.,
```

Comment:
a working tree?

Reply:
I think this means that if you were to check out that commit id (ie: git checkout abc123doremi) and build, it would build, and the functionality in the commit message would be available

Reply:
haha i meant he was missing the word 'a' in the sentence